common/include/pcl/experimental/executor/best_fit.hpp

```diff
@@ -0,0 +1,157 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2020-, Open Perception, Inc.
+ *  Author(s): Shrijit Singh <shrijitsingh99@gmail.com>
+ *
+ */
+
+#pragma once
+
+#include <pcl/experimental/executor/cuda_executor.hpp>
+#include <pcl/experimental/executor/inline_executor.hpp>
+#include <pcl/experimental/executor/omp_executor.hpp>
+#include <pcl/experimental/executor/sse_executor.hpp>
+#include <pcl/experimental/executor/type_trait.h>
+
+#include <boost/algorithm/string.hpp>
+#include <iostream>
+#include <thread>
+
+namespace executor {
+
+static const auto best_fit_executors =
+    std::make_tuple(executor::omp_executor<>{}, executor::sse_executor<>{},
+                    executor::inline_executor<>{});
+
+struct executor_runtime_checks {
+  template <typename Executor, typename executor::InstanceOf<
+                                   Executor, executor::inline_executor> = 0>
+  static bool check(Executor& exec) {
+    pcl::utils::ignore(exec);
+    return true;
+  }
+
+  template <typename Executor,
+            typename executor::InstanceOf<Executor, executor::sse_executor> = 0>
+  static bool check(Executor& exec) {
+    pcl::utils::ignore(exec);
+    if (const char* env_p = std::getenv("PCL_ENABLE_SSE_EXEC"))
+      return boost::iequals(env_p, "ON");
```

Comment:
this is not a boost worthy case. I was fine with what you had before.

Reply:
I switched because of this https://github.com/shrijitsingh99/pcl/pull/12#discussion_r476331089
I can switch back, if needed.

Reply:
`strcasecmp` is [POSIX extension](https://en.wikipedia.org/wiki/C_string_handling#Popular_extensions). Please keep boost usage.